tests/VObject/Recur/RRuleIteratorTest.php

```diff
@@ -922,6 +922,23 @@ public function testYearlyBySetPosLoop(): void
         );
     }
 
+    /**
+     * This caused an incorrect date to be returned by the rule iterator when
+     * start date was not on the rrule list.
+     */
+    public function testYearlyStartDateNotOnRRuleList(): void
+    {
+        $this->parse(
+            'FREQ=YEARLY;BYMONTH=6;BYDAY=-1FR;UNTIL=20250901T000000Z',
+            '2023-09-01 12:00:00',
+            [
+                '2023-09-01 12:00:00',
```

Comment:
Yes, having the first day not following the RRule pattern is a violation of the RFC, so maybe a validation error is needed somewhere...

Reply:
> Yes, having the first day not following the RRule pattern is a violation of the RFC, so maybe a validation error is needed somewhere...

We are already being flexible and allowing the first/start day to not have to be a match for the RRULE pattern. And the code schedules that start day. So I suppose we had better keep that behavior, otherwise there would be a BC break.

Reply:
I like the flexibility here, so +1